src/components/backup/RestoreCloudStep.tsx

```diff
@@ -71,38 +86,47 @@ const Masthead = styled(Box).attrs({
 });
 
 const KeyboardSizeView = styled(KeyboardArea)({
-  backgroundColor: ({ theme: { colors } }: any) => colors.transparent,
+  backgroundColor: ({ theme: { colors } }: ComponentProps) => colors.transparent,
 });
 
 type RestoreCloudStepParams = {
   RestoreSheet: {
-    selectedBackup: Backup;
+    selectedBackup: BackupFile;
   };
 };
 
 export default function RestoreCloudStep() {
   const { params } = useRoute<RouteProp<RestoreCloudStepParams & RestoreSheetParams, 'RestoreSheet'>>();
+  const { password } = backupsStore(state => ({
+    password: state.password,
+  }));
 
-  const { userData } = useCloudBackups();
+  const loadingState = walletLoadingStore(state => state.loadingState);
 
   const { selectedBackup } = params;
   const { isDarkMode } = useTheme();
-  const [loading, setLoading] = useState(false);
+  const { canGoBack, goBack } = useNavigation();
+
+  const onRestoreSuccess = useCallback(() => {
+    while (canGoBack()) {
+      goBack();
+    }
+  }, [canGoBack, goBack]);
 
   const dispatch = useDispatch();
+  const profilesEnabled = useExperimentalFlag(PROFILES);
   const { width: deviceWidth, height: deviceHeight } = useDimensions();
-  const { replace, navigate, getState: dangerouslyGetState, goBack } = useNavigation();
   const [validPassword, setValidPassword] = useState(false);
   const [incorrectPassword, setIncorrectPassword] = useState(false);
-  const [password, setPassword] = useState('');
   const passwordRef = useRef<TextInput | null>(null);
   const initializeWallet = useInitializeWallet();
 
   useEffect(() => {
     const fetchPasswordIfPossible = async () => {
-      const pwd = await fetchBackupPassword();
+      const pwd = await getLocalBackupPassword();
       if (pwd) {
-        setPassword(pwd);
+        backupsStore.getState().setStoredPassword(pwd);
+        backupsStore.getState().setPassword(pwd);
```

Comment:
If this store gets persisted (haven't checked) this is not a good idea

Reply:
it is not persisted